### PR TITLE
JDK-6285888: ChoiceFormat can support unescaped relational symbols in the Format segment

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this PR and [CSR](https://bugs.openjdk.org/browse/JDK-8314822) which allows for the _Format_ segment of a ChoiceFormat pattern to use the ChoiceFormat Relational symbols without the need to escape them using quotes. Previously, using a non escaped Relational symbol within a _Format_ segment would throw an IAE.

Implementation wise, this can be achieved by adding an additional check of `part == 0` to the relational symbol conditional. This is safe to do, as any subsequent relational symbols should only come after a '|' is parsed, which sets part back to 0. This ensures that for the duration of `part = 1` (Format segment), the relational symbols can be used without syntactic meaning.

For example, this change allows behavior such as: `new ChoiceFormat("1#The code is #7281")`. Previously, the workaround would have been new `ChoiceFormat("1#The code is '#'7281")` to achieve the same formatted behavior. Please see the CSR for more detail.

Additionally, this change also includes an unrelated grouping of the ChoiceFormat examples under a single header: **Usage Information**.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8314822](https://bugs.openjdk.org/browse/JDK-8314822) to be approved

### Issues
 * [JDK-6285888](https://bugs.openjdk.org/browse/JDK-6285888): ChoiceFormat can support unescaped relational symbols in the Format segment (**Bug** - P4)
 * [JDK-8314822](https://bugs.openjdk.org/browse/JDK-8314822): ChoiceFormat can support unescaped relational symbols in the Format segment (**CSR**)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17640/head:pull/17640` \
`$ git checkout pull/17640`

Update a local copy of the PR: \
`$ git checkout pull/17640` \
`$ git pull https://git.openjdk.org/jdk.git pull/17640/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17640`

View PR using the GUI difftool: \
`$ git pr show -t 17640`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17640.diff">https://git.openjdk.org/jdk/pull/17640.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17640#issuecomment-1917918413)

### CODE REVIEW

File: src/java.base/share/classes/java/text/ChoiceFormat.java [hunk 1]
```diff
@@ -172,20 +119,51 @@
...
  * </dl>
  *
  * </blockquote>
  *
  * <i>Note:The relation &le; is not equivalent to &lt;&equals;</i>
  *
- * <p>If a <i>Relation</i> symbol is to be used within a <i>Format</i> pattern,
- * it must be single quoted. For example,
- * {@code new ChoiceFormat("1# '#'1 ").format(1)} returns {@code " #1 "}.
- * Use two single quotes in a row to produce a literal single quote. For example,
```
Comment (naotoj) [inline src/java.base/share/classes/java/text/ChoiceFormat.java:185]: It sounds right to remove this restriction, but if we were to use `|` inside `Format`, shouldn't it still need to be single-quoted?

Comment (justin-curtis-lu) [reply src/java.base/share/classes/java/text/ChoiceFormat.java]: Right the restriction shouldn't be removed, but rather updated. Fixed in https://github.com/openjdk/jdk/pull/17640/commits/740686b2052a276b77529bea5444692e5bbe0078

**Review by naotoj — APPROVED** [review]
LGTM

**Pull request closed without merging**